### PR TITLE
CACTUS-765 Reach-less Modal

### PR DESCRIPTION
**Pull request by wilysword**
https://repayonline.atlassian.net/browse/CACTUS-765

The biggest things Reach did were scroll trapping (although they did it in a really weird way that broke our layout), focus lock, and handling visibility/events. We already had scroll trapping code, a focus lock, and the Dimmer that controls visibility; that just left the events, which are quite simple. The only really new thing here is the portal.

Also, there's one thing they did which I didn't, because it seemed like overkill: they set `aria-hidden` on everything _but_ the open modal.

### CODE REVIEW

**Review by Dhalton — APPROVED** [review]
Surprisingly much simpler than I would've imagined. I published this with `yalc` and tested in the report delivery app to check that this also gets rid of the issue with the body & html height being set to 0 and it works great.

File: modules/cactus-web/src/IconButton/IconButton.tsx [hunk 1]
```diff
@@ -160,6 +160,22 @@ const IconButtonBase = React.forwardRef<HTMLButtonElement, ButtonProps>(
   }
 )
 
+const focusOutline = system({
```
Comment (maxrcollins) [inline modules/cactus-web/src/IconButton/IconButton.tsx:163]: Is this just a rewrite or did you need to change a behavior in these styles?

Comment (wilysword) [reply modules/cactus-web/src/IconButton/IconButton.tsx:163]: The behavior is the same, but this makes it work with responsive values (`iconSize={['small', 'medium']}`). The old code directly indexed the mapping with the prop value, so if the prop was an array you'd end up with a style like `height: calc(100% + undefinedpx)`, which obviously doesn't work.

Comment (maxrcollins) [reply modules/cactus-web/src/IconButton/IconButton.tsx:163]: Got it. Thanks!

**Review by maxrcollins — APPROVED** [review]
Nicely done 👍

File: modules/cactus-web/src/helpers/portal.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { ReactNode, ReactPortal, useRef } from 'react'
+import { createPortal } from 'react-dom'
+
+const ID = 'cactus-portal-root'
```
Comment (maxrcollins) [inline modules/cactus-web/src/helpers/portal.ts:4]: Is it a bad idea to use a UUID or something on the off chance that we eventually want 2 portals on a page?

Comment (maxrcollins) [reply modules/cactus-web/src/helpers/portal.ts:4]: Oh wait, this gets thrown into the ether as soon as the modal is closed doesn't it? Nevermind then

Comment (wilysword) [reply modules/cactus-web/src/helpers/portal.ts:4]: That's not how `createPortal` works. There's only one root, and any number of portals can be added to it. Try adding this to a story file, for example:
```
import { usePortal } from '../helpers/portal'
const MyPortal = (props) => usePortal(<div {...props} />)
export const Portals = () => (
  <MyPortal id="one">
    <MyPortal id="two" />
  </MyPortal>
)

```

Comment (wilysword) [reply modules/cactus-web/src/helpers/portal.ts:4]: Also, although Reach's portal root is removed when not in use, I saw no harm in keeping this one around permanently.

Comment (maxrcollins) [reply modules/cactus-web/src/helpers/portal.ts:4]: I see. Thanks

**Pull request merged**